ArchiSteamFarm/GitHub.cs

```diff
@@ -0,0 +1,132 @@
+﻿//     _                _      _  ____   _                           _____
+//    / \    _ __  ___ | |__  (_)/ ___| | |_  ___   __ _  _ __ ___  |  ___|__ _  _ __  _ __ ___
+//   / _ \  | '__|/ __|| '_ \ | |\___ \ | __|/ _ \ / _` || '_ ` _ \ | |_  / _` || '__|| '_ ` _ \
+//  / ___ \ | |  | (__ | | | || | ___) || |_|  __/| (_| || | | | | ||  _|| (_| || |   | | | | | |
+// /_/   \_\|_|   \___||_| |_||_||____/  \__|\___| \__,_||_| |_| |_||_|   \__,_||_|   |_| |_| |_|
+// 
+// Copyright 2015-2018 Łukasz "JustArchi" Domeradzki
+// Contact: JustArchi@JustArchi.net
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading.Tasks;
+using ArchiSteamFarm.Localization;
+using Newtonsoft.Json;
+
+namespace ArchiSteamFarm {
+	internal static class GitHub {
+		internal static async Task<List<ReleaseResponse>> GetNLatestReleases(uint n) {
+			if(n == 0) {
+				ASF.ArchiLogger.LogNullError(nameof(n));
+				return null;
+			}
+
+			string releaseURL = SharedInfo.GithubReleaseURL + "?per_page=" + n;
+
+			return await GetReleasesFromURL(releaseURL).ConfigureAwait(false);
+		}
+
+		internal static async Task<ReleaseResponse> GetLatestRelease(bool stable = true) {
+			string releaseURL = SharedInfo.GithubReleaseURL + (stable ? "/latest" : "?per_page=1");
+
+			if (stable) {
+				return await GetReleaseFromURL(releaseURL).ConfigureAwait(false);
+			}
+			else {
+				return (await GetReleasesFromURL(releaseURL).ConfigureAwait(false))?.ElementAt(0);
+			}
+		}
+
+		internal static async Task<ReleaseResponse> GetRelease(string version) {
+			if(string.IsNullOrEmpty(version)) {
+				ASF.ArchiLogger.LogNullError(nameof(version));
+				return null;
+			}
+
+			return await GetReleaseFromURL(SharedInfo.GithubReleaseURL + "/tags/" + version).ConfigureAwait(false);
+		}
+
+		private static async Task<ReleaseResponse> GetReleaseFromURL(string releaseURL) {
+			if (string.IsNullOrEmpty(nameof(releaseURL))) {
+				ASF.ArchiLogger.LogNullError(nameof(releaseURL));
+				return null;
+			}
+
+			WebBrowser.ObjectResponse<ReleaseResponse> objectResponse = await Program.WebBrowser.UrlGetToJsonObject<ReleaseResponse>(releaseURL).ConfigureAwait(false);
+			if (objectResponse?.Content == null) {
+				ASF.ArchiLogger.LogNullError(nameof(objectResponse) + " || " + nameof(objectResponse.Content));
```

Comment:
You're logging null error on remote failures, this is invalid, error is for us and needs correction, remote failure does NOT need correction from us. Remove logging here.